woocommerce/woocommerce
# Monorepo Utils: Migrate cli-core

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Closes https://github.com/woocommerce/woocommerce/issues/37947

Migrate the `tools/cli-core` project to `@woocommerce/monorepo-utils` package. There is some de-duplication of utility functions and moving some functions into the `core/` folder. Since this PR simply moves files around and does not touch logic, I think checking to see if the commands compile is adequate test, but feel free to fully test the commands.

This PR did not clean up the use of Octokit and different methods of authentication. I will leave that for a follow up PR.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. `pnpm install`
2. Check the verify-day command
```
pnpm run utils code-freeze verify-day
```
3. Check the milestone command (dry run)
```
pnpm run utils code-freeze milestone -d
```
4. Check the branch command (dry run)
```
pnpm run utils code-freeze branch -d
```
5. Check the release-posts compiles
```
cd tools/release-posts && pnpm run release-post release --help
```
6. Check the Code Analyzer compiles
```
cd ../code-analyzer && pnpm run analyzer --help
```
7. Check the version bump tool compiles
```
cd ../version-bump && pnpm run version bump --help
```
